src/core/ColorUtil.cc

```diff
@@ -100,4 +103,194 @@ std::optional<Color4f> parse_hex_color(const std::string& hex) {
   return rgba;
 }
 
-} // namespace
\ No newline at end of file
+std::optional<Color4f> parse_web_color(const std::string& col) {
+  std::string colorname = boost::algorithm::to_lower_copy(col);
+  if (webcolors.find(colorname) != webcolors.end()) {
+    return webcolors.at(colorname);
+  } 
+  return {};
+}
+
+std::optional<Color4f> parse_color(const std::string& col) {
+  const auto webcolor = parse_web_color(col);
+  if (webcolor) {
+    return webcolor;
+  }
+
+  const auto hexcolor = parse_hex_color(col);
+  if (hexcolor) {
+    return hexcolor;
+  }
+
+  return {};
+}
+
+Color4f getColor(const std::string& col, const Color4f& defaultcolor)
+{
+  const auto parsed = parse_color(col);
+
+  if (!parsed) {
+    LOG(message_group::Warning, "Unable to parse color \"%1$s\", reverting to default color.", col);
+    LOG("Please see https://en.wikipedia.org/wiki/Web_colors");
+  } 
+
+  return parsed.value_or(defaultcolor);
+}
+
+// Colors extracted from https://drafts.csswg.org/css-color/ on 2015-08-02
+// CSS Color Module Level 4 - Editor’s Draft, 29 May 2015
+std::unordered_map<std::string, Color4f> webcolors{
```

Comment:
Once you separate the public from the private API, you can move the private functions, as well as this map, into an anonymous namespace

Reply:
Not sure if I moved this correctly.

Reply:
By "anonymous namespace" I meant create a block like this:
```
namespace {

std::unordered_map<std::string, Color4f> webcolors{ [ your stuff here ] };

std::optional<Color4f> parse_web_color(const std::string& col) { ...}

etc.

}  // namespace
```

That way, all of those implementations will be taken off the global namespace and only be available within this file. This is a very common way of hiding implementation details in C++